### PR TITLE
Introduced dashboard page and models for A-Level

### PR DESCRIPTION
**Pull request by Caleb-T-Owens**
## Status

* Current Status: Ready for (front-end / tech) review
* Review App link(s):
- https://teachcomputing-pr-1768.herokuapp.com/certificate/a-level
- https://teachcomputing-pr-1768.herokuapp.com/dashboard
* Closes https://github.com/NCCE/teachcomputing.org-issues/issues/2449
* Closes https://github.com/NCCE/teachcomputing.org-issues/issues/2448
* Closes https://github.com/NCCE/teachcomputing.org-issues/issues/2450 (bar mailers which have been moved to a new ticket)
## Review progress:

- [ ] Browser tested
- [ ] Front-end review completed
- [x] Tech review completed

## What's changed?

- Implemented A level certificate models and dashboard page
- Implemented public page
- Implemented Complete page

As we don't have the mailers yet, they've not been implemented. The class marker quiz is also not ready and in another ticket

### CODE REVIEW

File: app/controllers/certificates/a_level_controller.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    before_action :find_programme, only: %i[show complete pending]
+    before_action :user_enrolled?, only: %i[show complete pending]
+    before_action :user_programme_enrolment_pending?, only: %i[show complete]
+    after_action :discourage_caching
+
+    def show
+      return redirect_to complete_a_level_path if @programme.user_completed?(current_user)
+
+      assign_achievements
+      @professional_development_groups = @programme.programme_activity_groupings.not_community.order(:sort_key)
```
Comment (dcorking) [inline app/controllers/certificates/a_level_controller.rb:14]: Should there be a list of enrolled+completed courses that aren't listed in the programme activity groupings?

**Review by dcorking — COMMENTED** [review]
Great piece of work. You can check on the small changes I mentioned and fix up the parts that are out of scope (re test handling) later.

File: app/controllers/certificates/a_level_controller.rb (hunk 1 above)
Comment (Caleb-T-Owens) [reply app/controllers/certificates/a_level_controller.rb:13]: No, it doesn't look like we currently show completed courses that are not recommended

File: app/views/certificates/a_level/_achievement_list.html.erb [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
       <div class='ncce-activity-list__button-wrapper'>
         <%= link_to t('.all_courses_button'),
-          courses_path(level: 'Key stage 5'),
+          courses_path(level: 'Post 16'),
```
Comment (dcorking) [inline app/views/certificates/a_level/_achievement_list.html.erb:15]: This doesn't look quite right as there will probably be courses in that link that aren't in the programmes_activities table. IMO it should match what the user gets when the pick the 'A Level' course filter themselves.
Screenshot 2023-09-19 at 16 04 31

Comment (dcorking) [reply app/views/certificates/a_level/_achievement_list.html.erb]: Oh dear, even the filter seems to depend on Smart Connector data. That said, in my opinion the Figma was misleading. I suggest you merge now and revisit when the courses are configured, but for consistency at least, I think the certificate category is more likely to be consistent with courses that will actually show up in the user's A Level dashboard as enrolled/completed.

Comment (Caleb-T-Owens) [reply app/views/certificates/a_level/_achievement_list.html.erb]: I do agree

**Review by dcorking — DISMISSED** [review]
LGTM 🚂

**Pull request merged**